code/modules/projectiles/projectile/magic.dm

```diff
@@ -154,6 +154,10 @@
 	qdel(src)
 
 /proc/wabbajack(mob/living/M, randomize)
+	if(istype(M, /mob/living/simple_animal/hostile/mimic/copy) || ispAI(M))
+		M.wabbajack_act()
+		return
```

Comment:
The idea was to remove the need of an `istype()` check at all.
You can typecast this whole proc on living, for example, and redefine it at the pAI and mimic level.

Reply:
Done with 638ee3ef7843486c6e698cf14f8948d29853128f